histograms/histogram.go

```diff
@@ -0,0 +1,83 @@
+package histograms
+
+import (
+	"sync"
+	"time"
+
+	"github.com/wavefronthq/wavefront-sdk-go/senders"
+
+	tdigest "github.com/caio/go-tdigest"
+)
+
+type Histogram struct {
+	mutex              sync.Mutex
+	priorTimedBinsList []*timedBin
+	currentTimedBin    *timedBin
+
+	Granularity time.Duration
+	Compression uint32
+	MaxBins     int
+}
+
+type timedBin struct {
+	tdigest   *tdigest.TDigest
+	timestamp time.Time
+}
+
+type Distribution struct {
+	Centroids []senders.Centroid
+	Timestamp time.Time
+}
+
+func NewHistogram() *Histogram {
```

Comment:
As commented above, take relevant values as input and return the `Histogram` interface.

Reply:
Refer to https://github.com/rcrowley/go-metrics/blob/master/histogram.go for example.